src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java

```diff
@@ -295,4 +296,22 @@ public Action updateAction(final Action action) {
     }
     return existingAction;
   }
+
+  @Transactional(
+      propagation = Propagation.REQUIRED,
+      readOnly = false,
+      timeout = TRANSACTION_TIMEOUT)
+  public void rerunAction(final Action action) throws CTPException {
```

Comment:
`CTPException` is one of my pet hates. Really what we're talking about is an `IllegalStateException` runtime exception if we attempt to transition from a state that we really should have known in advance wasn't going to work. The most useful thing we can return from our REST API is 403 - bad request, I think.

Consider catching the `CTPException` and then throwing the appropriate runtime exception, so we can get rid of all the `throws` checked exception declarations more easily in future.

Reply:
:+1: to this but bad request should be 400 status code right? 403 is unauthorized